common/core/mosaicdisplay.cpp

```diff
@@ -542,13 +609,96 @@ bool MosaicDisplay::ContainConnector(const uint32_t connector_id) {
 #ifdef ENABLE_PANORAMA
 void MosaicDisplay::SetPanoramaMode(bool mode) {
   panorama_mode_ = mode;
+  if (panorama_mode_)
+    panorama_enabling_state_ = true;
 }
 
-void MosaicDisplay::SetExtraDispInfo(int num_physical_displays,
-                                     int num_virtual_displays) {
-  num_physical_displays_ = num_physical_displays;
-  num_virtual_displays_ = num_virtual_displays;
+void MosaicDisplay::SetExtraDispInfo(
+    DisplayManager *displaymgr_ptr,
+    std::vector<NativeDisplay *> *virtual_panorama_displays,
+    std::vector<NativeDisplay *> *physical_panorama_displays) {
+  displaymgr_ptr_ = displaymgr_ptr;
+  virtual_panorama_displays_ = virtual_panorama_displays;
+  physical_panorama_displays_ = physical_panorama_displays;
+
+  num_physical_displays_ = physical_panorama_displays->size();
+
+  num_virtual_displays_ = virtual_panorama_displays->size();
+
+  for (uint32_t i = 0; i < physical_panorama_displays->size(); i++) {
+    real_physical_displays_.emplace_back(physical_panorama_displays->at(i));
+  }
+}
+
+bool MosaicDisplay::EnablePanorama() {
+  if (!panorama_mode_)
+    return false;
+
+  if (!panorama_enabling_state_) {
```

Comment:
may need to lock to protect this panorama_enabling_state_
suppose this function is in the HWC.info binder thread
and obvious this state_ should be protected as functions in other thread may need to access this too
So this state need to be protected by lock

Reply:
Okay, will revise the code pieces to have panorama_enabling_state_ lock protected.